pyproject.toml

```diff
@@ -51,8 +51,8 @@ markers = ["slow", "installation_test"]
 requires = ["setuptools>=42.0.0",
             "setuptools_scm",
             "wheel",
-            "oldest-supported-numpy",
```

Comment:
@bsipocz this PR led by @AlanPearl establishes compatibility between halotools and numpy 2.0. However, I'm not sure about this particular line of code because I never really understood what using `oldest-supported-numpy` did. But this PR fails unless this line is changed to `numpy>=2.0`.

On a possibly related note, I'm also not sure whether any additional changes need to be made in order to properly use `extension-helpers` in the right way.